UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AttributePopUp.cs

```diff
@@ -1,18 +1,59 @@
-﻿using UnityEngine;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using UnityEngine;
 using TMPro;
 using UnityEngine.UI;
 using AnimArch.Visualization.Diagrams;
+using Attribute = AnimArch.Visualization.Diagrams.Attribute;
 
 namespace AnimArch.Visualization.UI
 {
     public class AttributePopUp : AbstractPopUp
     {
         public TMP_Dropdown dropdown;
         public Toggle isArray;
+        public TMP_Text confirm;
+        private readonly HashSet<TMP_Dropdown.OptionData> _variableData = new();
+        private string _formerAttributeName;
 
-        public void ActivateCreation(TMP_Text classTxt, TMP_Text atrTxt)
+        private void UpdateDropdown()
+        {
+            var classNames = DiagramPool.Instance.ClassDiagram.GetClassList().Select(x => x.Name);
+
+            dropdown.options.RemoveAll(x => _variableData.Contains(x));
+            _variableData.Clear();
+            _variableData.UnionWith(classNames.Select(x => new TMP_Dropdown.OptionData(x)));
+            dropdown.options.AddRange(_variableData);
+        }
+
+        public override void ActivateCreation(TMP_Text classTxt)
+        {
+            base.ActivateCreation(classTxt);
+            UpdateDropdown();
+            confirm.text = "Add";
+        }
+
+
+        public void ActivateCreation(TMP_Text classTxt, TMP_Text attributeTxt)
         {
             ActivateCreation(classTxt);
+            var text = attributeTxt.text.Split(": ");
+            var formerName = text[0];
+
+            var attributeType = text[1];
+            var formerArray = attributeType.Contains("[]");
+            attributeType = Regex.Replace(attributeType, "[\\[\\]\\n]", "");
+            var formerType = attributeType;
+
+            isArray.isOn = formerArray;
+            inp.text = formerName;
+            dropdown.value = dropdown.options.FindIndex(x => x.text == formerType);
+
+            _formerAttributeName = formerName;
+
+            confirm.text = "Edit";
```

Comment:
```suggestion
            ActivateCreation(classTxt);
            var text = attributeTxt.text.Split(": ");

            var attributeType = text[1];
            attributeType = Regex.Replace(attributeType, "[\\[\\]\\n]", "");

            isArray.isOn = attributeType.Contains("[]");
            inp.text = text[0];
            dropdown.value = dropdown.options.FindIndex(x => x.text == attributeType);

            _formerAttributeName = text[0];

            confirm.text = "Edit";
```
+ nemoze nastat pri ristupe na index 0 a 1 access violation?

Reply:
ako vsade, vsetko by malo mat typ